pkg/lightning/backend/kv/session.go

```diff
@@ -61,19 +60,18 @@ func (b *bytesBuf) add(v []byte) []byte {
 	start := b.idx
 	copy(b.buf[start:], v)
 	b.idx += len(v)
-	return b.buf[start:b.idx]
+	return b.buf[start:b.idx:b.idx]
 }
 
 func newBytesBuf(size int) *bytesBuf {
 	return &bytesBuf{
-		buf: manual.New(size),
+		buf: make([]byte, size),
```

Comment:
CGO is used here to better manage memory, could you do a benchmark to make sure replace CGO with go slice won't increase the peek memory usage

Reply:
CGO is not actually used here because the `manual` dependency is `br/pkg/manual` 😂.

Reply:
But the bytesBuffer in pkg/lightning/backend/local depends on pkg/lightning/manual with uses CGO, so I think we'd better do some benchmark or just keep the manual implement.

Reply:
okay, I will do a benchmark later

Reply:
After benchmark lightning local backend without `manual.go`, I found the peak memory increased by about 30%. So I will consider adding `manual.go` back. 

However, BR doesn't need CGO. It's hard to keep a same implementation  of  `byteBuffer`, since there is only one `bytesRecycleChan`. But for this work to continue, I think we can introduce an `Allocator` interface to allocate and free memory. The code looks like:
```go
type Allocator interface {
	Alloc(n int) []byte
	Free([]byte)
}

type stdAllocator struct{}

func (stdAllocator) Alloc(n int) []byte {
	return make([]byte, n)
}

func (stdAllocator) Free(b []byte) {}

type Pool struct {
	ch        chan []byte
	allocator Allocator
}

func NewPool(size int, allocator Allocator) *Pool {
	return &Pool{
		ch:        make(chan []byte, size),
		allocator: allocator,
	}
}

func (p *Pool) Acquire() []byte {
	select {
	case b := <-p.ch:
		return b
	default:
		return p.allocator.Alloc(1 << 20) // 1M
	}
}

func (p *Pool) Release(b []byte) {
	select {
	case p.ch <- b:
	default:
		p.allocator.Free(b)
	}
}

func (p *Pool) NewBuffer() *Buffer {
	return &Buffer{pool: p, bufs: make([][]byte, 0, 128), curBufIdx: -1}
}

var globalPool = NewPool(1024, stdAllocator{})

var NewBuffer = globalPool.NewBuffer
```

What do you think about it? @glorv

Reply:
LGTM